javagen/src/main/java/com/azure/autorest/template/EnumTemplate.java

```diff
@@ -195,4 +202,16 @@ protected String createEnumJsonCreatorIfCheck(EnumType enumType) {
             return String.format("item.%s().equals(value)", toJsonMethodName);
         }
     }
+
+    protected void addGeneratedImport(Set<String> imports) {
+        if (JavaSettings.getInstance().isDataPlaneClient()) {
```

Comment:
Do we need this check? We should enable partial-update for non-DPG clients as well. But if we don't have this supported for clients and client builders, we can defer this to later. Eventually, all DPG and non-DPG generations should support partial updates. 
(No change for this PR but likely a GitHub issue to follow up).

Reply:
Vanilla also supports @Generated for `clients` and `builders`,  I add this check after [discussing](https://github.com/Azure/autorest.java/issues/2036#issuecomment-1519567045) with Weidong to reduce the impact,I can remove this check and test in next pr.

Reply:
This can be done in a separate PR but please file an issue to track this.

Reply:
Issue created: https://github.com/Azure/autorest.java/issues/2130